### PR TITLE
chore: api auth state design doc

### PR DESCRIPTION
**Pull request by decleaver**
## Description
Update for considering handling api auth state in backend (using cookies) vs frontend

### CODE REVIEW

**Comment by UncleGedd** [comment]
Few recommendations:
- delete any empty sections
- make it clear which approach we are using (I believe it's the one currently under "Alternative Considered")
- IMO there's no need to detail out the previous approach because it will be deprecated soon

**Comment by TristanHoladay** [comment]
working on [integrate authsvc PR](https://github.com/defenseunicorns/uds-runtime/pull/335) and wondering if we should work in-cluster auth into this design doc as well.

**Comment by UncleGedd** [comment]
> working on [integrate authsvc PR](https://github.com/defenseunicorns/uds-runtime/pull/335) and wondering if we should work in-cluster auth into this design doc as well.

IMO for now we should keep this design doc focused on desktop auth

File: design-docs/0001-api-auth.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+### Implementation Details
+The API uses a token-based authentication system. The token is generated by the backend server and is used to authenticate the user. API authentication is enabled by default, to disable it you can set the `API_AUTH_DISABLED` environment variable to true.
+
+Instead of managing state in the frontend, the api authentication state is managed in the backend. One way this can be accomplished is by using session cookies.
+
+The generated api token is used initially by the frontend to authenticate with the backend, but once the token is validated, the backend generates a session ID and sets the session ID as a secure, HttpOnly cookie in the response. The frontend receives the response and the browser automatically stores the session cookie. For subsequent requests, the browser automatically includes the session cookie and the backend authMiddleware extracts and validate the session ID from the cookie. This approach simplifies the frontend codebase and reduces the amount of state management needed in the frontend.
+
+How does the frontend authenticate?
```
Comment (UncleGedd) [inline design-docs/0001-api-auth.md:17]:
```suggestion
The authentication flow is as follows:
```

File: design-docs/0001-api-auth.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Author(s): @decleaver  
+Date Created: Sept 17, 2024  
+Status: APPROVED
+
+### Problem Statement
+API authentication is needed to prevent unauthorized access to the API from other processes when running UDS Runtime locally.
+
+### Implementation Details
+The API uses a token-based authentication system. The token is generated by the backend server and is used to authenticate the user. API authentication is enabled by default, to disable it you can set the `API_AUTH_DISABLED` environment variable to true.
```
Comment (UncleGedd) [inline design-docs/0001-api-auth.md:11]: I keep getting hung up on `API_AUTH_DISABLED`, IMO it's easier to read the positive, as in `API_AUTH_ENABLED`. I think @TristanHoladay brought up something similar in a previous PR?

Comment (decleaver) [reply design-docs/0001-api-auth.md:11]: yeah, it used to be `API_AUTH_ENABLED` but I switched it to `API_AUTH_DISABLED` to emphasize that the default state is enabled based on comments Jeff had made. So if the env var is not set, api auth is enabled. You only really need to set this env when you want to disable it.

Comment (UncleGedd) [reply design-docs/0001-api-auth.md:11]: 👍 I'm good to keep it, appreciate the reasoning!

**Review by UncleGedd — APPROVED** [review]

**Comment by UncleGedd** [comment]
One last thing, need to update the name of the files becase `0001-` already exists in the `design-docs` folder

**Comment by UncleGedd** [comment]
Almost there! Just need to de-dedup

Screenshot 2024-09-18 at 3 39 22 PM

**Review by UncleGedd — APPROVED** [review]

**Pull request merged**